src/modules/peek/Peek.FilePreviewer/FilePreview.xaml.cs

```diff
@@ -127,6 +150,13 @@ private async Task UpdatePreviewAsync(CancellationToken cancellationToken)
                     var size = await Previewer.GetPreviewSizeAsync(cancellationToken);
                     SizeFormat windowSizeFormat = UnsupportedFilePreviewer != null ? SizeFormat.Percentage : SizeFormat.Pixels;
                     PreviewSizeChanged?.Invoke(this, new PreviewSizeChangedArgs(size, windowSizeFormat));
+
+                    if (Previewer is ImagePreviewer imagePreviewer && ScalingFactor != 0)
```

Comment:
Is it possible to keep the logic out of the FilePreview control since it only affect IBitmapPreviewer? I think it would be cleaner to keep specific previewer logic out of the FilePreview control if possible.

Reply:
I've pushed the update, please let me know if it's better now.